package/kedro_viz/models/flowchart.py

```diff
@@ -278,24 +304,36 @@ def has_metadata(self) -> bool:
         return self.kedro_obj is not None
 
 
-@dataclass
-class GraphNodeMetadata(abc.ABC):
+class GraphNodeMetadata(BaseModel, abc.ABC):
```

Comment:
Why do you need to inherit from `abc`?

Reply:
I kept the original signature, but added BaseModel to it. Though we do not have any fields in this class, this is supposed to be an abstract base class.